ERROR in Illegal state: symbol without members expected when using AOT in 4.3.0

## I'm submitting a...
<pre><code>
[x] Regression
[ ] Bug report 
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior
Getting error when building for production (AOT enabled) with a certain pattern of imports/exports (see below). The error reported is:
```
ERROR in Illegal state: symbol without members expected, but got {"filePath":"snip/test/ng/aottest/src/app/shared/role-co
nstants.ts","name":"RoleConstants","members":["USER"]}.

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in 'snip\test\ng\aottest\src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts
```
Worked ok in 4.2.6, does not work in 4.3.0.

## Expected behavior
ng build --prod should build correctly without rising the error.

## Minimal reproduction of the problem with instructions
download [symbol-without-members-error-test-a.zip](https://github.com/angular/angular/files/1152729/symbol-without-members-error-test-a.zip)
npm install
ng build --prod

The project contain a newly generated project (with ng new) with a number of files added to reproduce the error:
shared/const.ts - exports a constant field
shared/index.ts - exports consts above
intermediate.ts - imports consts exports aggregated data
index.ts - exports intermediate above
module.ts - is calling RouterModule.forRoot([...intermediate] with intermediate imported from above.

If `import { Consts } from './shared';` is changed to `import { Consts } from './shared/consts';` in intermediate.ts the problem does not manifest anymore.

## What is the motivation / use case for changing the behavior?
n/a

## Environment

<pre><code>
Angular version: 4.3.0
(worked ok in 4.2.6.)

For Tooling issues:
- Node version: v7.10.0
- Platform: Windows

</code></pre>

--- a/packages/compiler/src/aot/static_reflector.ts
+++ b/packages/compiler/src/aot/static_reflector.ts
@@ -415,7 +415,9 @@ export class StaticReflector implements CompileReflector {
           for (const item of (<any>expression)) {
             // Check for a spread expression
             if (item && item.__symbolic === 'spread') {
-              const spreadArray = simplify(item.expression);
+              // We call with references as 0 because we require the actual value and cannot
+              // tolerate a reference here.
+              const spreadArray = simplifyInContext(context, item.expression, depth, 0);
               if (Array.isArray(spreadArray)) {
                 for (const spreadItem of spreadArray) {
                   result.push(spreadItem);
@@ -432,9 +434,10 @@ export class StaticReflector implements CompileReflector {
           return result;
         }
         if (expression instanceof StaticSymbol) {
-          // Stop simplification at builtin symbols or if we are in a reference context
+          // Stop simplification at builtin symbols or if we are in a reference context and
+          // the symbol doesn't have members.
           if (expression === self.injectionToken || self.conversionMap.has(expression) ||
-              references > 0) {
+              (references > 0 && !expression.members.length)) {
             return expression;
           } else {
             const staticSymbol = expression;

--- a/packages/compiler/test/aot/static_reflector_spec.ts
+++ b/packages/compiler/test/aot/static_reflector_spec.ts
@@ -878,6 +878,42 @@ describe('StaticReflector', () => {
     });
   });
 
+  // Regression #18170
+  it('should continue to aggresively evaluate enum member accessors', () => {
+    const data = Object.create(DEFAULT_TEST_DATA);
+    const file = '/tmp/src/my_component.ts';
+    data[file] = `
+      import {Component} from '@angular/core';
+      import {intermediate} from './index';
+
+      @Component({
+        template: '<div></div>',
+        providers: [{provide: 'foo', useValue: [...intermediate]}]
+      })
+      export class MyComponent { }
+    `;
+    data['/tmp/src/intermediate.ts'] = `
+      import {MyEnum} from './indirect';
+      export const intermediate = [{
+        data: {
+          c: [MyEnum.Value]
+        }
+      }];`;
+    data['/tmp/src/index.ts'] = `export * from './intermediate';`;
+    data['/tmp/src/indirect.ts'] = `export * from './consts';`;
+    data['/tmp/src/consts.ts'] = `
+      export enum MyEnum {
+        Value = 3
+      }
+    `;
+    init(data);
+
+    expect(reflector.annotations(reflector.getStaticSymbol(file, 'MyComponent'))[0]
+               .providers[0]
+               .useValue)
+        .toEqual([{data: {c: [3]}}]);
+  });
+
 });
 
 const DEFAULT_TEST_DATA: {[key: string]: any} = {

